services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -1082,10 +1082,16 @@ public static AuthResult verifyIdentityToken(KeycloakSession session, RealmModel
                 }
             }
 
-            UserModel user = session.users().getUserById(token.getSubject(), realm);
```

Comment:
I'm wondering how pairwise was working without this change. It seems that token verification would always fail. Is my understanding correct ?

Reply:
I'm not sure that part of the code is involved when verifying refresh tokens. But you're right. It's odd.